docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,364 @@
+      * [Creation](#creation)
+      * [Execution](#execution)
+      * [Deletion](#deletion)
+    * [Use cases](#use-cases)
+      * [Recovery from permanent quorum loss](#recovery-from-permanent-quorum-loss)
+      * [Trigger on-demand snapshot compaction](#trigger-on-demand-snapshot-compaction)
+      * [Trigger on-demand full/delta snapshot](#trigger-on-demand-fulldelta-snapshot)
+      * [Trigger on-demand maintenance of etcd cluster](#trigger-on-demand-maintenance-of-etcd-cluster)
+      * [Copy backups from one object store to another object store](#copy-the-backups-from-one-object-store-to-another-object-store)
+  * [Metrics](#metrics)
+
+## Summary
+
+This DEP proposes an enhancement to `etcd-druid`'s capabilities to handle [out-of-band](#terminology) tasks, which are presently performed manually or invoked programmatically via sub-optimal APIs. The document proposes the establishment of an unified interface by defining a well-structured API to harmonize the initiation of any `out-of-band` task, monitor its status, and simplify the process of adding new tasks and managing their lifecycle.
+
+## Terminology
+
+* **druid:** [etcd-druid](https://github.com/gardener/etcd-druid) is an operator to manage the etcd clusters.
+
+* **backup-sidecar:** It is the etcd-backup-restore sidecar container running in each etcd-member pod of etcd cluster.
+
+* **leading-backup-sidecar:** A backup-sidecar that is associated to an etcd leader of an etcd cluster.
+
+* **out-of-band task** Any on-demand tasks/operations that can be executed on an etcd cluster without modifying the Etcd custom resource spec (desired state).
+
+## Motivation
+
+Today, [Etcd-druid](https://github.com/gardener/etcd-druid) mainly acts as an etcd cluster provisioner(creation, maintenance and deletion). In future, capabilities of etcd-druid will be enhanced via [etcd-member](https://github.com/gardener/etcd-druid/blob/8ac70d512969c2e12e666d923d7d35fdab1e0f8e/docs/proposals/04-etcd-member-custom-resource.md) proposal by providing it access to a much more detailed information about each etcd cluster member. While we enhance the reconciliation and monitoring capabilities of etcd-druid, it still lacks an ability to provide an easy-to-use hook to trigger `out-of-band` tasks on an existing etcd cluster.
+
+There are new learnings while operating etcd clusters at scale. It has been observed that we regularly need capabilities to trigger `out-of-band` tasks which are out of the purview of a regular etcd reconciliation run. Many of these tasks are a multi-step process and doing it manually is error-prone even if an operator follows a well-written step-by-step guide. Thus there is a need to automate these tasks to alleviate mistakes.
+Some examples of an `on-demand/out-of-band` operations:
+
+* Recover from a permament quorum loss.
+* Trigger an on-demand full/delta snapshot.
+* Trigger an on-demand snapshot compaction.
+* Trigger an on-demand maintenance of etcd cluster.
```

Comment:
I'm guessing you are talking about supporting the operation mentioned in the upstream doc here.
```suggestion
* Trigger an on-demand [maintenance of etcd cluster](https://etcd.io/docs/v3.4/op-guide/maintenance/).
```

Reply:
I have moved this link to where we have defined/described each operations.